internal/jobservice/eventstojobs/eventstojobs.go

```diff
@@ -38,20 +39,28 @@ func NewEventsToJobService(
 	}
 }
 
-// Subscribes to a JobSet from jobsetid
+// Subscribes to a JobSet from jobsetid. Will retry until there is a successful exit.
 func (eventToJobService *EventsToJobService) SubscribeToJobSetId(context context.Context) error {
-	return eventToJobService.streamCommon(&eventToJobService.jobServiceConfig.ApiConnection, context)
+	for {
+		err := eventToJobService.streamCommon(&eventToJobService.jobServiceConfig.ApiConnection, context)
+		if err == nil {
+			return nil
+		} else {
+			time.Sleep(time.Second * 5)
+		}
+	}
 }
 
 func (eventToJobService *EventsToJobService) streamCommon(clientConnect *client.ApiConnectionDetails, ctx context.Context) error {
 	var fromMessageId string
-	conn, connErr := client.CreateApiConnection(clientConnect)
-	if connErr != nil {
-		log.Warnf("Connection Issues with EventClient %v", connErr)
-		return connErr
-	}
-	defer conn.Close()
+	var conn *grpc.ClientConn
+	defer func() {
+		if conn != nil {
+			conn.Close()
+		}
+	}()
 	eventToJobService.jobServiceRepository.SubscribeJobSet(eventToJobService.queue, eventToJobService.jobSetId)
+	ctx, cancel := context.WithCancel(ctx)
```

Comment:
Should we be mixing cancelling context and error groups here?

Reply:
I thought about that but error group doesn't give you access to the cancel function which we kind of need in order to cause a no-error termination

Reply:
What I'm reading -- the parent context CancelFunc will also cancel the children/derived contexts. The inverse is not true. In this case, I think it's okay -- we use the errgroup to capture the goroutine return error but we're not relying on it's cancellation.